### ISSUE TITLE
Graceful handling when ItemsSource is empty

### ISSUE THREAD

**Issue by jfversluis** [markolazic88/SwipeCardView#7]
Just added your (amazing) package to a new project and get something started. It looks and works really great, thank you for that!

To get started, I basically copied your Tinder example and start getting all the snippets I needed into my own project. I just forgot to set the BindingContext on my page which caused the items to be empty. Of course, totally on me. But when I swiped, it would give me an NRE and the whole app crashed. It might be nice to go about that a little more graceful. Here is the strack trace:

```
2018-12-05 08:52:23.706968+0100 SwipeConf.iOS[9516:7626443] 
Unhandled Exception:
System.NullReferenceException: Object reference not set to an instance of an object
  at MLToolkit.Forms.SwipeCardView.SwipeCardView.OnPanUpdated (System.Object sender, Xamarin.Forms.PanUpdatedEventArgs e) [0x00006] in <69a7c63c842440d492feb3399307c652>:0 
  at Xamarin.Forms.PanGestureRecognizer.SendPanStarted (Xamarin.Forms.Element sender, System.Int32 gestureId) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\PanGestureRecognizer.cs:38 
  at Xamarin.Forms.Platform.iOS.EventTracker+<>c__DisplayClass15_0.<GetNativeRecognizer>b__2 (UIKit.UIPanGestureRecognizer r) [0x000a2] in <18e93b810a4a4e2f919016f742528d37>:0 
  at UIKit.UIPanGestureRecognizer+Callback.Activated (UIKit.UIPanGestureRecognizer sender) [0x00000] in /Library/Frameworks/Xamarin.iOS.framework/Versions/12.2.1.11/src/Xamarin.iOS/UIKit/UIGestureRecognizer.cs:229 
--- End of stack trace from previous location where exception was thrown ---
  at (wrapper managed-to-native) UIKit.UIApplication.UIApplicationMain(int,string[],intptr,intptr)
  at UIKit.UIApplication.Main (System.String[] args, System.IntPtr principal, System.IntPtr delegate) [0x00005] in /Library/Frameworks/Xamarin.iOS.framework/Versions/12.2.1.11/src/Xamarin.iOS/UIKit/UIApplication.cs:79 
  at UIKit.UIApplication.Main (System.String[] args, System.String principalClassName, System.String delegateClassName) [0x0002c] in /Library/Frameworks/Xamarin.iOS.framework/Versions/12.2.1.11/src/Xamarin.iOS/UIKit/UIApplication.cs:63 
  at SwipeConf.iOS.Application.Main (System.String[] args) [0x00001] in /Users/jfversluis/Projects/SwipeConf/SwipeConf.iOS/Main.cs:17
2018-12-05 08:52:23.707868+0100 SwipeConf.iOS[9516:7626443] Unhandled managed exception:
Object reference not set to an instance of an object (System.NullReferenceException)
  at MLToolkit.Forms.SwipeCardView.SwipeCardView.OnPanUpdated (System.Object sender, Xamarin.Forms.PanUpdatedEventArgs e) [0x00006] in <69a7c63c842440d492feb3399307c652>:0 
  at Xamarin.Forms.PanGestureRecognizer.SendPanStarted (Xamarin.Forms.Element sender, System.Int32 gestureId) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\PanGestureRecognizer.cs:38 
  at Xamarin.Forms.Platform.iOS.EventTracker+<>c__DisplayClass15_0.<GetNativeRecognizer>b__2 (UIKit.UIPanGestureRecognizer r) [0x000a2] in <18e93b810a4a4e2f919016f742528d37>:0 
  at UIKit.UIPanGestureRecognizer+Callback.Activated (UIKit.UIPanGestureRecognizer sender) [0x00000] in /Library/Frameworks/Xamarin.iOS.framework/Versions/12.2.1.11/src/Xamarin.iOS/UIKit/UIGestureRecognizer.cs:229 
--- End of stack trace from previous location where exception was thrown ---
  at (wrapper managed-to-native) UIKit.UIApplication.UIApplicationMain(int,string[],intptr,intptr)
  at UIKit.UIApplication.Main (System.String[] args, System.IntPtr principal, System.IntPtr delegate) [0x00005] in /Library/Frameworks/Xamarin.iOS.framework/Versions/12.2.1.11/src/Xamarin.iOS/UIKit/UIApplication.cs:79 
  at UIKit.UIApplication.Main (System.String[] args, System.String principalClassName, System.String delegateClassName) [0x0002c] in /Library/Frameworks/Xamarin.iOS.framework/Versions/12.2.1.11/src/Xamarin.iOS/UIKit/UIApplication.cs:63 
  at SwipeConf.iOS.Application.Main (System.String[] args) [0x00001] in /Users/jfversluis/Projects/SwipeConf/SwipeConf.iOS/Main.cs:17
2018-12-05 08:52:23.708208+0100 SwipeConf.iOS[9516:7626443] critical: Stacktrace:
2018-12-05 08:52:23.708315+0100 SwipeConf.iOS[9516:7626443] critical: 
Native stacktrace:
2018-12-05 08:52:23.723877+0100 SwipeConf.iOS[9516:7626443] critical: 	0   SwipeConf.iOS                       0x00000001049f1cd4 mono_handle_native_crash + 244
2018-12-05 08:52:23.724097+0100 SwipeConf.iOS[9516:7626443] critical: 	1   libsystem_platform.dylib            0x0000000112accb3d _sigtramp + 29
2018-12-05 08:52:23.724223+0100 SwipeConf.iOS[9516:7626443] critical: 	2   ???                                 0x000000000000ffff 0x0 + 65535
2018-12-05 08:52:23.724314+0100 SwipeConf.iOS[9516:7626443] critical: 	3   libsystem_c.dylib                   0x000000011284fc45 abort + 127
2018-12-05 08:52:23.724433+0100 SwipeConf.iOS[9516:7626443] critical: 	4   SwipeConf.iOS                       0x0000000104bfeccf xamarin_unhandled_exception_handler + 47
2018-12-05 08:52:23.724550+0100 SwipeConf.iOS[9516:7626443] critical: 	5   SwipeConf.iOS                       0x0000000104a8f18e mono_invoke_unhandled_exception_hook + 158
2018-12-05 08:52:23.724660+0100 SwipeConf.iOS[9516:7626443] critical: 	6   SwipeConf.iOS                       0x00000001049f167c mono_handle_exception_internal + 6140
2018-12-05 08:52:23.724756+0100 SwipeConf.iOS[9516:7626443] critical: 	7   SwipeConf.iOS                       0x00000001049efe79 mono_handle_exception + 25
2018-12-05 08:52:23.724864+0100 SwipeConf.iOS[9516:7626443] critical: 	8   SwipeConf.iOS                       0x0000000104970343 mono_amd64_throw_exception + 131
2018-12-05 08:52:23.724963+0100 SwipeConf.iOS[9516:7626443] critical: 	9   ???                                 0x000000013399b5a7 0x0 + 5160678823
2018-12-05 08:52:23.725065+0100 SwipeConf.iOS[9516:7626443] critical: 	10  SwipeConf.iOS                       0x00000001048d323d _ZL32native_to_managed_trampoline_192P11objc_objectP13objc_selectorPP11_MonoMethodP22UIPanGestureRecognizerj + 461
2018-12-05 08:52:23.725232+0100 SwipeConf.iOS[9516:7626443] critical: 	11  SwipeConf.iOS                       0x00000001048d3068 -[__UIPanGestureRecognizer target:] + 56
2018-12-05 08:52:23.725413+0100 SwipeConf.iOS[9516:7626443] critical: 	12  UIKitCore                           0x000000011adb7b81 -[UIGestureRecognizerTarget _sendActionWithGestureRecognizer:] + 57
2018-12-05 08:52:23.725597+0100 SwipeConf.iOS[9516:7626443] critical: 	13  UIKitCore                           0x000000011adc0a5f _UIGestureRecognizerSendTargetActions + 109
2018-12-05 08:52:23.725838+0100 SwipeConf.iOS[9516:7626443] critical: 	14  UIKitCore                           0x000000011adbe244 _UIGestureRecognizerSendActions + 305
2018-12-05 08:52:23.725983+0100 SwipeConf.iOS[9516:7626443] critical: 	15  UIKitCore                           0x000000011adbd49c -[UIGestureRecognizer _updateGestureWithEvent:buttonEvent:] + 858
2018-12-05 08:52:23.726143+0100 SwipeConf.iOS[9516:7626443] critical: 	16  UIKitCore                           0x000000011adaef20 _UIGestureEnvironmentUpdate + 1329
2018-12-05 08:52:23.726317+0100 SwipeConf.iOS[9516:7626443] critical: 	17  UIKitCore                           0x000000011adae9ad -[UIGestureEnvironment _deliverEvent:toGestureRecognizers:usingBlock:] + 478
2018-12-05 08:52:23.726527+0100 SwipeConf.iOS[9516:7626443] critical: 	18  UIKitCore                           0x000000011adae71d -[UIGestureEnvironment _updateForEvent:window:] + 200
2018-12-05 08:52:23.726791+0100 SwipeConf.iOS[9516:7626443] critical: 	19  UIKitCore                           0x000000011b21e78a -[UIWindow sendEvent:] + 4058
2018-12-05 08:52:23.726989+0100 SwipeConf.iOS[9516:7626443] critical: 	20  UIKitCore                           0x000000011b1fc394 -[UIApplication sendEvent:] + 352
2018-12-05 08:52:23.727213+0100 SwipeConf.iOS[9516:7626443] critical: 	21  UIKitCore                           0x000000011b2d15a9 __dispatchPreprocessedEventFromEventQueue + 3054
2018-12-05 08:52:23.727354+0100 SwipeConf.iOS[9516:7626443] critical: 	22  UIKitCore                           0x000000011b2d41cb __handleEventQueueInternal + 5948
2018-12-05 08:52:23.727508+0100 SwipeConf.iOS[9516:7626443] critical: 	23  CoreFoundation                      0x00000001052f3721 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 17
2018-12-05 08:52:23.727637+0100 SwipeConf.iOS[9516:7626443] critical: 	24  CoreFoundation                      0x00000001052f2f93 __CFRunLoopDoSources0 + 243
2018-12-05 08:52:23.727770+0100 SwipeConf.iOS[9516:7626443] critical: 	25  CoreFoundation                      0x00000001052ed63f __CFRunLoopRun + 1263
2018-12-05 08:52:23.728013+0100 SwipeConf.iOS[9516:7626443] critical: 	26  CoreFoundation                      0x00000001052ece11 CFRunLoopRunSpecific + 625
2018-12-05 08:52:23.728148+0100 SwipeConf.iOS[9516:7626443] critical: 	27  GraphicsServices                    0x0000000115cc31dd GSEventRunModal + 62
2018-12-05 08:52:23.728286+0100 SwipeConf.iOS[9516:7626443] critical: 	28  UIKitCore                           0x000000011b1e081d UIApplicationMain + 140
2018-12-05 08:52:23.728499+0100 SwipeConf.iOS[9516:7626443] critical: 	29  ???                                 0x00000001379a347e 0x0 + 5227820158
2018-12-05 08:52:23.728696+0100 SwipeConf.iOS[9516:7626443] critical: 	30  ???                                 0x00000001379a3213 0x0 + 5227819539
2018-12-05 08:52:23.728870+0100 SwipeConf.iOS[9516:7626443] critical: 
=================================================================
Got a SIGABRT while executing native code. This usually indicates
a fatal error in the mono runtime or one of the native libraries 
used by your application.
=================================================================
```

It seems that I am still allowed to swipe, even while there are no items and then it causes a `NullReferenceException` because there are no items to work with.

**Comment by markolazic88** [comment]
Gerald, I am glad you like it! Your suggestion makes perfect sense. I will definitely work on handling this issue in a more graceful way.

**Comment by markolazic88** [comment]
This is finally included in 2.1.0 release.
